[LINK-2174] add support for authnMethod config (#110)
moneytree/mt-link-javascript-sdk: docs/README.md

| options.cobrandClientId (private)          | string                                      | false    |               | <strong>NOTE: This is an internal attribute. Please do not use it unless instructed by your integration representative.</strong><br /><br />Brand Moneytree apps with client's branding. E.g: logo or theme.                                                                                                                                                                                                                 |
| options.samlSubjectId   | string    | false    |   | Set subject Id for saml session version.                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

### setSamlSubjectId

> Only includes the parameter when calling authorize, logout, openService methods

Reviewer: Lol, I was expecting more explanation of what the method for instead of just using my note for the other half of the explanation.

The `setSamlSubjectId` parameter allows the client to pass a guest identifier to Moneytree so that Moneytree can forward it to the Identity Provider (IdP) via the SAMLRequest.

> 

Reviewer: should we also add which API we are going to pass this value to?

Author: It would make sense to have it

Author: added

```javascript
mtLinkSdk.setSamlSubjectId(samlSubjectId);
```

| Parameter                                  | Type                                        | Required | Default Value | Description                                                                                                                                                                                                                                                                                                                                                                                                                  |
| ------------------------------------------ | ------------------------------------------- | -------- | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | sublSubjectId                                   | string                                      | true     |               | <p>Overrides the saml_subject_id parameter</p>                                                                                                                                                                                                                 |

Reviewer: ```suggestion
| samlSubjectId                                   | string                                      | true     |               | <p>Set the saml_subject_id parameter</p>                                                                                                                                                                                                                 |
```

It can't overwrite if no one pass it via the init option.

| options.authnMethod   | `passwordless` or `sso`    | `undefined`    | Value set during `init`.<p><strong>OR</strong></p>`undefined` | Switch between authentication methods version.                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

> | options.sdkPlatform (private)                                     | string            | false    | Generated by the SDK.                                     | <strong>NOTE: this is for Moneytree internal use, please do not use it to avoid unintended behavior!</strong><br /><br />Indicating sdk platform.                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer 2: We also need the `samlSubjectId`

| options.authnMethod   | array of string with values `passwordless` or `sso` or string with value `passwordless` or `sso`    | `undefined`    | Value set during `init`.<p><strong>OR</strong></p>`undefined` | Switch between authentication methods version.                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

> | options.samlSubjectId   | string    | `undefined`    | Value set during `init`.<p><strong>OR</strong></p>`undefined` | Set subject Id for saml session version.                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: I don't think this belongs here, this sections is a configs parameters.

| options.showAuthToggle                                            | boolean           | false    | Value set during `init`.<p><strong>OR</strong></p>`true`  | If you wish to disable the login to sign up form toggle button and vice-versa in the auth screen, set this to `false`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |

> | options.showRememberMe                                            | boolean           | false    | Value set during `init`.<p><strong>OR</strong></p>`true`  | If you wish to disable the `Stay logged in for 30 days` checkbox in the login screen, set this to `false`. w                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |

Reviewer 2: ```suggestion
| options.showRememberMe                                            | boolean           | false    | Value set during `init`.<p><strong>OR</strong></p>`true`  | If you wish to disable the `Stay logged in for 30 days` checkbox in the login screen, set this to `false`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```

| options.isNewTab                                                  | boolean           | false    | Value set during `init`.<p><strong>OR</strong></p>`false` | Call method and open/render in a new browser tab, by default all views open in the same tab.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |

> | options.authnMethod   | array of string with values `passwordless` or `sso` or string with value `passwordless` or `sso`    | `undefined`    | Value set during `init`.<p><strong>OR</strong></p>`undefined` | Switch between authentication methods version.                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| options.authnMethod   | string <p><strong>OR</strong></p> string[] | | Value set during `init`. | Use different authentication methods. This can be a string or an array of the following values:<br /><br />`sso`, `passwordless`. |
```

Please stick to the existing document language or the way it explains for each column.

This options is also needed for some other APIs.

Author: we updated the docs to reflect the document language as you suggested here. What do you mean by `This option is also needed for some other APIs`??

Reviewer: Nope, my mistake, I thought you only add it to one of the API, just scroll up and saw it is under common API, all good.
